transaction-impl/src/main/java/com/example/auction/transaction/impl/TransactionRepository.java

```diff
@@ -0,0 +1,195 @@
+                                "WHERE (creatorId = ? OR winnerId = ?) AND status = ? " +
+                                "ORDER BY itemId DESC " +
+                                "LIMIT ?",
+                        userId,
+                        userId,
+                        status,
+                        limit
+                )
+                .thenApply(List::stream)
+                .thenApply(rows -> rows.skip(offset))
+                .thenApply(rows -> rows.map(TransactionRepository::toTransactionSummary))
+                .thenApply(transactionSummaries -> transactionSummaries.collect(Collectors.toList()))
+                .thenApply(TreePVector::from);
+    }
+
+    private static TransactionSummary toTransactionSummary(Row transaction) {
+        return new TransactionSummary(
+                transaction.getUUID("itemId"),
+                transaction.getUUID("creatorId"),
+                transaction.getUUID("winnerId"),
+                transaction.getString("itemTitle"),
+                transaction.getString("currencyId"),
+                transaction.getInt("itemPrice"),
+                transaction.get("status", TransactionInfoStatus.class)
+        );
+    }
+
+    private static class TransactionEventProcessor extends ReadSideProcessor<TransactionEvent> {
+        private final CassandraSession session;
+        private final CassandraReadSide readSide;
+
+        private PreparedStatement insertTransactionStatement;
+
+        @Inject
+        public TransactionEventProcessor(CassandraSession session, CassandraReadSide readSide) {
+            this.session = session;
+            this.readSide = readSide;
+        }
+
+        @Override
+        public ReadSideHandler<TransactionEvent> buildHandler() {
+            return readSide.<TransactionEvent>builder("transactionEventOffset")
+                    .setGlobalPrepare(this::createTable)
+                    .setPrepare(tag -> prepareStatements())
+                    .setEventHandler(TransactionEvent.TransactionStarted.class,
+                            e -> insertTransaction(e.getItemId(), e.getTransaction()))
+                    .build();
+        }
+
+        @Override
+        public PSequence<AggregateEventTag<TransactionEvent>> aggregateTags() {
+            return TransactionEvent.TAG.allTags();
+        }
+
+        private CompletionStage<Done> createTable() {
+            return session.executeCreateTable(
+                    "CREATE TABLE IF NOT EXISTS transactionSummary (" +
+                            "itemId timeuuid PRIMARY KEY, " +
+                            "creatorId UUID, " +
+                            "winnerId UUID, " +
+                            "itemTitle text, " +
+                            "currencyId text, " +
+                            "itemPrice int, " +
+                            "status text" +
+                            ") " +
+                            "WITH CLUSTERING ORDER BY (itemId DESC)"
```

Comment:
> Show all the transactions in `web-gateway`, and hide transactions in the UI depending on the filter.

@yg-apaza, I had in mind that the UI would show all transactions that I have been a part of, either as the seller or as the winner. To be honest, I hadn't even thought about filtering them, but just displaying them all together, maybe with different formatting for sold items vs. bought items. We haven't really defined very clear requirements for this... it could go either way. :smile:

Maybe for simplicity's sake, let's display all of the transactions in a single unfiltered list for now, since that makes things easier. We might choose to add filtering, or split the page in two, as a future enhancement.

With Cassandra, each type of query you want to do requires a separate table structure (with the possibility of using ["materialized views"](http://docs.datastax.com/en/cql/3.3/cql/cql_using/useCreateMV.html) that automatically derive one table from the data in another). This is one of the ways it differs from relational databases such as MySQL and PostgreSQL... Cassandra requires ["denormalization"](http://www.datastax.com/dev/blog/basic-rules-of-cassandra-data-modeling), or storing the same data in multiple ways. This is usually considered very bad practice for relational databases, partly because it makes things more complicated and it's easy to have bugs that cause the data to fall out of sync. However with Cassandra, this denormalization is what allows it to be so fast and scalable. Lagom's event sourcing read-side design helps to ensure that nothing falls out of sync.

So, this means that the data design needs to closely reflect the way the data is used in the application. For example, if we wanted to have separate queries to get transactions where I am the seller, and transactions where I am the winner, it would require two different tables with two different primary keys (`transactionsByWinner` and `transactionsBySeller`). The read-side processor would need to insert into both. If we wanted to get all transactions mixed together with an optional `userType` filter, we would need to create a `userType` column in the table, and make it part of the primary key.